### ISSUE TITLE
Confusing 'by hostname' view when containers have their own hostnames

### ISSUE THREAD

**Issue by 2opremio** [weaveworks/scope#826]
I don't know if this is intentional and I simply don't understand the "by hostname" view" but, the "by hostname" view of my kubernetes cluster includes the hostnames from the containers, which I find confusing:

<img width="1323" alt="screen shot 2016-01-15 at 5 08 28 pm" src="https://cloud.githubusercontent.com/assets/2362916/12359557/e3ad2b1e-bbaa-11e5-8e68-3c9c1c43bf8a.png">

```
$ docker ps  | grep app-mapper | grep -v pause
faf115d00c15        weaveworks/app-mapper-db                             "/docker-entrypoint.s"   3 days ago          Up 3 days                               k8s_app-mapper-db.3281346e_app-mapper-db-jimdf_default_10d549d8-b93d-11e5-a532-0800279b9f1d_9d3d0a02
$ docker exec -ti faf115d00c15 bash
root@app-mapper-db-jimdf:/# hostname
app-mapper-db-jimdf
```

**Comment by tomwilkie** [comment]
Thats the idea - containers by (their) hostname.  Its useful if you use weave dns for simple load balancing.  It doesn't look useful for k8s.

**Comment by 2opremio** [comment]
Should I close this then?

**Comment by tomwilkie** [comment]
Can you think of a way to make this less confusing?  What were you expecting?

**Comment by 2opremio** [comment]
I was expecting to only see hostnames from container-running hosts (i.e. hostnames of machines running Docker).

**Comment by 2opremio** [comment]
However, I do understand that it has value to show the hostnames of containers so I don't know what I would change. Maybe the view name? I don't have better suggestions though.

**Comment by tomwilkie** [comment]
We could say "By Container Hostname" but it might be too wordy.  Or "By Their Hostname" but that seems a little informal.  Lets see if anyone else has any ideas.

**Comment by peterbourgon** [comment]
I just raised this question independently in the chat.

> I was expecting to only see hostnames from container-running hosts (i.e. hostnames of machines running Docker).

I also expect this. (Although then I would expect the label to be "By host", and TBH I don't really see the utility of it.)

> Thats the idea - containers by (their) hostname. 

I've always understood the hostname of a container to be basically arbitrary. That containers launched with WeaveDNS encodes a sort of "role" there is nice, but exceptional. Am I wrong there?

**Comment by tomwilkie** [comment]
> I've always understood the hostname of a container to be basically arbitrary. That containers launched with WeaveDNS encodes a sort of "role" there is nice, but exceptional. Am I wrong there?

I think there will be no one-size fits all.  In docker right now, they definitely aren't arbitrary (docker seems to be moving away from links and toward hostnames in /etc/hosts and dns as a way of connecting containers).  In k8s they might be, not sure.

I had hoped that by putting this under the "containers" heading, the hostname would implicitly be the container hostname.  Perhaps we should change the label to "by dns name"?

**Comment by 2opremio** [comment]
> I had hoped that by putting this under the "containers" heading, the hostname would implicitly be the container hostname.

Really good point. Haven't thought about that but, for some reason the _By Host_ view immediately makes me think about the hosts running the containers. I think the term `host` may have an implicit meaning of machine.

> Perhaps we should change the label to "by dns name"?

I cannot think about any better names but perhaps it's not only a name but also a presentation (UI) problem. Hosts and host-processes get the same node-representation as containers and even the internet node. I think that having a different node shape/icon for different types of nodes/views would help a lot here.

**Comment by tomwilkie** [comment]
> I think that having a different node shape/icon for different types of nodes/views would help a lot here.

I agree.  @foot had some ideas here that he put together into a wireframe:

<img width="695" alt="screen shot 2016-01-24 at 14 19 30" src="https://cloud.githubusercontent.com/assets/444037/12539344/84be2412-c2a5-11e5-964d-6d7df9c9cbba.png">

Rest of the wireframes are here: https://docs.google.com/presentation/d/1grDt76fIGo32G5pARQssgkZUceEvUkN1XNk1GV96NjI/edit#slide=id.gdb85c3591_0_464

Thats being tracked by #659, so I suggest we continue the discussion re:icons there.
